### PR TITLE
mon: a few more upmap (and other) fixes

### PR DESCRIPTION
**Pull request by xiexingguo**

### CODE REVIEW

File: src/osd/OSDMap.cc [hunk 1]
```diff
@@ -3789,6 +3789,10 @@ int OSDMap::calc_pg_upmaps(
 		     << " pgs " << pgs << dendl;
     }
 
+    if (osd_weight_total == 0) {
+      lderr(cct) << __func__ << "abort due to osd_weight_total == 0" << dendl;
```
Comment (tchaikov) [inline src/osd/OSDMap.cc]: a space before "abort"

**Comment by idryomov** [comment]
@liewegas Queued it up, thanks!

File: qa/workunits/cephtool/test.sh [hunk 2]
```diff
@@ -1831,8 +1831,10 @@ function test_mon_pg()
   expect_false ceph osd primary-affinity osd.9999 .5
   ceph osd primary-affinity osd.0 1
 
-  ceph osd pg-temp 1.0 0 1 2
-  ceph osd pg-temp 1.0 osd.1 osd.0 osd.2
+  ceph osd pool set rbd size 2
+  ceph osd pg-temp 1.0 0 1
+  ceph osd pg-temp 1.0 osd.1 osd.0
+  expect_false ceph osd pg-temp 1.0 0 1 2
```
Comment (xiexingguo) [inline qa/workunits/cephtool/test.sh]: @liewegas Sorry, 
http://qa-proxy.ceph.com/teuthology/sage-2017-07-10_16:55:37-rados-wip-sage-testing-distro-basic-smithi/1383107/teuthology.log 
http://qa-proxy.ceph.com/teuthology/sage-2017-07-10_16:55:37-rados-wip-sage-testing-distro-basic-smithi/1383043/teuthology.log

The above two test failures are related to this change. I think the above change fixes these now.

**Comment by xiexingguo** [comment]
retest this please

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-07-11_14:43:46-rados-wip-sage-testing-distro-basic-smithi/

**Pull request merged**